### PR TITLE
Add functions to assign/unassign private IP addresses

### PR DESCRIPTION
**Pull request by s-sebastian**
### What does this PR do?

Add functions to assign/unassign private IP addresses on AWS EC2 instance
### What issues does this PR fix or reference?

N/A
### Tests written?

No

### CODE REVIEW

File: salt/modules/boto_ec2.py [hunk 1]
```diff
@@ -422,6 +422,118 @@ def disassociate_eip_address(public_ip=None, association_id=None, region=None,
...
+        (bool)   - True on success, False on failure.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion boto_ec2.assign_private_ip_addresses network_interface_name=my_eni private_ip_addresses=private_ip
+        salt myminion boto_ec2.assign_private_ip_addresses network_interface_name=my_eni secondary_private_ip_address_count=2
+
+    .. versionadded:: 2016.3.0
```
Comment (rallytime) [inline salt/modules/boto_ec2.py]: Can you change this to `Nitrogen` instead of `2016.3.0`?

File: salt/modules/boto_ec2.py [hunk 2]
```diff
@@ -422,6 +422,118 @@ def disassociate_eip_address(public_ip=None, association_id=None, region=None,
...
+    returns
+        (bool)   - True on success, False on failure.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion boto_ec2.unassign_private_ip_addresses network_interface_name=my_eni private_ip_addresses=private_ip
+
+    .. versionadded:: 2016.3.0
```
Comment (rallytime) [inline salt/modules/boto_ec2.py]: Same comment here.

**Comment by rallytime** [comment]
@s-sebastian This looks good to me. Thanks for adding these! I just had a couple of small changes that should be fixed up before we can get this in.

@techhat Do you have anything to add?

**Comment by s-sebastian** [comment]
@rallytime version has been updated as requested.

I've also noticed that `if network_interface_name...` statement returns unicode type for interface id:

```
<?xml version="1.0" encoding="UTF-8"?>
<Response><Errors><Error><Code>InvalidNetworkInterfaceId.Malformed</Code><Message>Invalid id: "{'result': u'eni-9d17c4d0'}" (expecting "eni-...")</Message></Error></Errors><RequestID>25cbb035-7102-400f-8764-967264df519c</RequestID></Response>
```

This also affects `associate_eip_address` function.

**Comment by cachedout** [comment]
@s-sebastian You could log those exceptions in your try/except blocks with `log.exception` if you like. Otherwise this looks good.

**Pull request merged**